PocketKit/Sources/PocketKit/Refresh/HomeRefreshCoordinator.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import UIKit
+import Combine
+import Sync
+
+protocol HomeRefreshCoordinatorProtocol {
+    func refresh(isForced: Bool, _ completion: @escaping () -> Void)
+}
+
+class HomeRefreshCoordinator: HomeRefreshCoordinatorProtocol {
+    static let dateLastRefreshKey = "HomeRefreshCoordinator.dateLastRefreshKey"
+    private let notificationCenter: NotificationCenter
+    private let userDefaults: UserDefaults
+    private let source: Source
+    private let refreshRuleTime: Int
+    private var subscriptions: [AnyCancellable] = []
+    private var isRefreshing: Bool = false
+
+    init(notificationCenter: NotificationCenter, userDefaults: UserDefaults, source: Source, refreshRuleTime: Int = 12) {
+        self.userDefaults = userDefaults
+        self.notificationCenter = notificationCenter
+        self.refreshRuleTime = refreshRuleTime
+        self.source = source
+
+        self.notificationCenter.publisher(for: UIScene.willEnterForegroundNotification, object: nil).sink { [weak self] _ in
+            self?.refresh { }
+        }.store(in: &subscriptions)
+    }
+
+    func refresh(isForced: Bool = false, _ completion: @escaping () -> Void) {
+        if shouldRefresh(isForced: isForced), !isRefreshing {
+            Task {
+                do {
+                    isRefreshing = true
+                    try await source.fetchSlateLineup(HomeViewModel.lineupIdentifier)
+                    userDefaults.setValue(Date(), forKey: Self.dateLastRefreshKey)
+                } catch {
+                    Crashlogger.capture(error: error)
```

Comment:
As a complement to capturing the error, would it be worth adding a breadcrumb in the `do` block to mark that a refresh has been made?

Reply:
Sounds good to me! I think it's valuable to understand how often the refresh occurs. Does this make sense for the breadcrumb details? 

`Crashlogger.breadcrumb(category: "refresh", level: .info, message: "Home Refresh Occur")`

Reply:
That works for me!